tubular/scripts/frontend_utils.py

```diff
@@ -106,11 +106,11 @@ def install_requirements_npm_aliases(self):
         """ Install NPM alias requirements for app to build """
         npm_aliases = self.get_npm_aliases_config()
         if npm_aliases:
-            # Install and pin NPM to latest npm@8 version
-            proc = subprocess.Popen(['npm install npm@8'], cwd=self.app_name, shell=True)
+            # Install and pin NPM to latest npm@10 version
+            proc = subprocess.Popen(['npm install npm@10'], cwd=self.app_name, shell=True)
```

Comment:
This feels like scope creep to me... @adamstankiewicz can you help me understand why we should pull in work that previously was unable to land due to unknown complexity?

Reply:
@jristau1984 Only the `npm ci` bit is scope creep (OK to defer as separate initiative, but shouldn't be forgotten/lost IMHO; that's the only reason I'm mentioning `npm ci` specifically).

The rest of the feedback/questions is still pertinent; relying on the supported Node/NPM versions defined by the `.nvmrc` file within each MFE itself is desired/preferable, without having a single code path override the global `npm` version, potentially to a version incompatible with the installed Node version.

Reply:
Thank you for the feedback. The goal of this PR is to pin `npm` to `v10` to address the immediate issues caused by the transition from `node` `v16` to `v18` and to ensure smooth compatibility as we work towards `node` `v20`. The suggestions about using `.nvmrc`, `nvm`, and switching to `npm ci` are definitely valid points that could help with consistency and future-proofing.

However, these changes introduce broader adjustments that I would say go beyond the immediate scope of this PR. Since they could require further investigation, I think it would make sense to handle this in a separate ticket. This way, we can thoroughly explore these improvements without delaying the `npm` pinning fix.

Reply:
> [question] Why can we install dependencies without an npm version override within `install_requirements`, but not in `install_requirements_npm_aliases`?

I would still love clarification/exploration on this question from above, before simply bumping the pinned version here. Do we really need to pin at all within `install_requirements_npm_aliases` when other install commands don't? Has this been explored yet?

My understanding is that `FrontendBuilder` should be relying on the Node version configured via the pipeline template that executed the `frontend_build.py` script. Is this not true?

Might it make sense to evaluate which NPM version is already available in this code path before simply bumping the pinned `npm` version to understand whether the pinning is even needed at all, given the other install commands don't pin? For example, could we log the current Node/NPM versions before pinning `npm` to see if we're actually already running the intended Node version here (sans pinning)?

---

Based on the [Git blame](https://github.com/openedx-unsupported/tubular/pull/370/files) for why this `npm install npm` is even within `install_requirements_npm_aliases` to begin with is due to when some builds were using NPM 6, which didn't support NPM aliases. Given we're well beyond NPM 6 at this point, there's a really good chance the pinning is just not needed anymore.

```js
# Install the latest version of npm that supports aliases (>= 6.9.0)
proc = subprocess.Popen(['npm install npm'], cwd=self.app_name, shell=True)
```

---

tl;dr; removing the `npm` pinning altogether _might_ have the same effect as pinning `npm` to latest `v10` version. Seems worthwhile to explore before simply proceeding with bumping the pinned `npm` version. I don't feel it's really much additional scope to explore this question...